iocore/hostdb/P_RefCountCache.h

```diff
@@ -239,44 +246,37 @@ template <class C>
 void
 RefCountCachePartition<C>::erase(uint64_t key, ink_time_t expiry_time)
 {
-  location_type l = this->item_map.find(key);
-  if (l.isValid()) {
-    if (expiry_time >= 0 && l.m_value->meta.expiry_time != expiry_time) {
+  if (auto it = this->item_map.find(key); it != this->item_map.end()) {
```

Comment:
Why is this a for loop? It seems without incrementing the iterator, this could in theory loop forever?